[CISupport] kill-old-processes: Do not kill Xvfb with number :0

lauromoura opened the pull request:
#### 917d04b25a60f35ed46515904a258e65c52a3edc
<pre>
[CISupport] kill-old-processes: Do not kill Xvfb with number :0
<a href="https://bugs.webkit.org/show_bug.cgi?id=247714">https://bugs.webkit.org/show_bug.cgi?id=247714</a>

Reviewed by Jonathan Bedard, Aakash Jain and Carlos Alberto Lopez Perez.

* Tools/CISupport/kill-old-processes:

Canonical link: <a href="https://commits.webkit.org/256569@main">https://commits.webkit.org/256569@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/17bf6c74c10e5a1939df1a20da773fb66267df6d

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/96167 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/5420 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/29221 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/105715 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/166052 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/100144 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/5544 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/34183 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/88550 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/101536 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/101826 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/4110 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/82773 "Built successfully") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/31139 "Passed tests") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/85957 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/87865 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/73968 "Found 2 new API test failures: /WebKitGTK/TestWebKitWebView:/webkit/WebKitWebView/is-web-process-responsive, /TestWTF:WTF_ParkingLot.UnparkOneOneFast (failure)") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/39908 "Built successfully") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/73/builds/19386 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/37582 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/20739 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/4550 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/5 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/60/builds/43341 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/5 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/40003 "Passed tests") | | 
<!--EWS-Status-Bubble-End-->

clopez commented on Tools/CISupport/kill-old-processes:
```diff
@@ -217,6 +216,9 @@ def main(user=None):
         removeOrphanShmSegments()
+
+        # Some bots use a Xvfb instance on :0 for dbus activation, so don't kill it.
+        os.system("ps aux | grep -P 'Xvfb' | grep -v ' :0' | grep -v grep | awk '{print $2}' | xargs -r kill")
```
This could still end killing the xvfb process that starts the bot main script. It is not guaranteed that the assigned screen to this first xvfb process is zero. It can have another number, specially if the service is restarted. We run the process with ```xvfb-run -a``` (the `-a` parameter means to assign to it automatically the first free screen that may be `0` or another number).

In order to differentiate from the `Xvfb` process launched by the test suite (the ones we want to kill) vs the ones launched by the bot startup helpers I suggest to grep also for the ```displayfd``` parameter that we pass on the test suite but is not passed on a normal usage of `Xvfb` with `xvfb-run`.

Suggested line:
`
 os.system("ps aux | grep -P 'Xvfb -displayfd [0-9]'  | awk '{print $2}' | xargs -r kill")`

Just a tip: when you want to filter grep itself from the output of grepping the process is enough with adding a regular expression to the expression grepped.
```ps aux | grep -P bash``` will show the grep process itself but ```ps aux | grep -P bas[h]``` will not. That works because with the regular expression the name of the process ```grep -P bas[h]``` not longer matches the regular expression (`bas[h]` as a literal string doesn't match `bas[h]` as a regular expression, but `bash` does)

lauromoura committed:
[CISupport] kill-old-processes: Do not kill Xvfb with number :0
https://bugs.webkit.org/show_bug.cgi?id=247714

Reviewed by Jonathan Bedard, Aakash Jain and Carlos Alberto Lopez Perez.

* Tools/CISupport/kill-old-processes:

Canonical link: https://commits.webkit.org/256569@main
